ls1intum/Artemis
# [Test] Exam management tests

### Motivation and Context


We want to add End-to-End tests to test exam functionality in Artemis.

### Description

This PR adds tests, which verify that the exam creation and deletion via the UI works. Furthermore the test adds functionality to create exams via request, so that other E2E tests can quickly generate exams in their setup phase. This will be the base for all other exam related E2E tests.

### Steps for Testing


1. Configure the adminUsername/adminPassword in `cypress.env.json` for your testsystem of choice
2. Set the baseUrl in `cypress.json` to the url of your testsystem of choice
3. Execute `yarn cypress:run --spec src/test/cypress/integration/ExamManagement.spec.ts` (Adjust / to \ if you are on Windows)

The bamboo buildplan will execute the tests against ts3, so it is best that you verify that the tests run there. I executed this PR with the [testing buildplan](https://bamboo.ase.in.tum.de/browse/ARTEMIS-AETBBT-130) and all tests passed. If you need help with configuring the test credentials feel free to write me.
